scripts/build/termux_step_setup_variables.sh

```diff
@@ -187,6 +187,9 @@ termux_step_setup_variables() {
 	TERMUX_PYTHON_HOME=$TERMUX_PREFIX/lib/python${TERMUX_PYTHON_VERSION} # location of python libraries
 	TERMUX_PKG_MESON_NATIVE=false
 	TERMUX_PKG_CMAKE_CROSSCOMPILING=true
+	TERMUX_PKG_ONLY_INSTALLING=false # when set to true you indicate that the package can only be installed
+	TERMUX_PKG_SEPARATE_SUB_DEPENDS=false # when set to true, subpackages will not be added to the dependency of the parent package
+	TERMUX_PKG_ACCEPT_PKG_IN_DEP=false # when set to true, the package can be in its own dependency list if necessary (useful for cyclic dependencies)
```

Comment:
By default, the `buildorder.py` logic does not add a package to the package's dependency list, even if there is a circular dependency that the package itself binds. But since `build-package.sh` can work with cyclic dependencies, and if the package itself is needed to compile the package (there are various reasons for this), then you can configure this variable.

For better understanding, here is an example of a conditional dependency list for compiling the `glib` package, which has a circular dependency with the `gobject-introspection` package:
```bash
# Result of the dependency list for compiling `glib` with value disabled:
...
gobject-introspection
...

# Result of the dependency list for compiling `glib` with value enabled:
...
glib
gobject-introspection
...
```

Reply:
Reopening, as above. A huge thing missing from this entire conversation (or from the higher-level one we need to have right after) is *why* circular dependencies are expected. Seems like they weren't before the glibc stuff came in, since `build-all.sh` used to work and the logic you're trying to replace handled them by failing.

Why is it correct and necessary that `glib -> gobject-introspection -> glib`? How would I know when it is and isn't appropriate to set this flag on any given other package?

Reply:
> Seems like they weren't before the glibc stuff came in, since build-all.sh used to work and the logic you're trying to replace handled them by failing.

This is not for `build-all.sh`, but for `build-package.sh` with `buildorder.py`.

> Why is it correct and necessary that glib -> gobject-introspection -> glib?

This question is off topic. Simple explanation: `gobject-introspection` needs `glib` to collect its files, and `glib` needs `gobject-introspection` to collect files from `gobject-introspection` (if I remember correctly).

> How would I know when it is and isn't appropriate to set this flag on any given other package?

To understand this, you need to try to build the package.